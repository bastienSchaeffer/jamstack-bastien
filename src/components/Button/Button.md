
# Button From MD

## Based on styled-system / emotion

**Styled systemed:**
  - borderRadius
  - buttonStyle
  - space
  - fontSize
  - variant

~~~js
  <Button
    onClick={() => setDialog(true)}
    size={'medium'}
    variant="danger">
    Click Here
  </Button>
~~~


      