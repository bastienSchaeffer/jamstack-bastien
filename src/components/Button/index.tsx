import styled from 'styled-components';
import {
  borderRadius,
  buttonStyle,
  space,
  fontSize,
  variant,
} from 'styled-system';

const buttonSize = variant({
  prop: 'size',
  key: 'buttonSizes',
});

const Button = styled('button')(
  {
    appearance: 'button',
    border: 0,
    outline: 0,
    boxShadow: '0 3px 10px rgba(0,0,0,.16)',
    transition: 'all 0.2s ease',
  },
  borderRadius,
  buttonStyle,
  buttonSize,
  space,
  fontSize
);

Button.defaultProps = {
  borderRadius: 2,
};

Button.displayName = `Button`;

export default Button;
