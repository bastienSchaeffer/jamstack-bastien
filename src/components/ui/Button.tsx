import styled, { css } from "styled-components";

export const Button = styled.button`
  background: rgba(120, 130, 140, 0.13);
  min-width: 7rem;
  padding: 0.5rem 1rem;
  font-size: 1rem;
  line-height: 1.25;
  color: ${({ theme }) => theme.colors.light};
  display: inline-block;
  font-weight: 500;
  outline: none;

  user-select: none;
  border: 1px solid transparent;
  padding: 0.5rem 1rem;
  border-radius: ${props => (props.round ? `500px` : `.25rem`)};
  margin: 0 0.2rem 0.2rem 0;

  :disabled {
    opacity: 0.4;
  }

  ${props =>
    props.primary &&
    css`
      background: ${({ theme }) => theme.colors.darkLight};
      color: ${({ theme }) => theme.colors.light};
      :hover {
        background: ${({ theme }) => theme.colors.darkDark};
      }
    `}
`;

export const ButtonDanger = styled(Button)`
  background: ${({ theme }) => theme.colors.dangerDark};
  background: red;
  :hover {
    background: ${({ theme }) => theme.colors.dangerLight};
  }
`;

export const ButtonWarn = styled(Button)`
  background: ${({ theme }) => theme.colors.warn};
  :hover {
    background: ${({ theme }) => theme.colors.warnLight};
  }
`;
