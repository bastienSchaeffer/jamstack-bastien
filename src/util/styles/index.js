import * as styledComponents from '@emotion/styled';
import { Global, css, keyframes } from '@emotion/core';
import { ThemeProvider } from 'emotion-theming';
import propTypes from '@styled-system/prop-types';
import { themeGet } from '@styled-system/theme-get';
import { theme } from './theme';

const { default: styled } = styledComponents;

export { css, Global, keyframes, ThemeProvider, theme, themeGet, propTypes };
export * from 'styled-system';
export default styled;
