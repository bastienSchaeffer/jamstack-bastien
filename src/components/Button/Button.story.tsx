import React from 'react';
import Button from './index';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import readme from './Button.md';
import {
  withKnobs,
  text,
  radios,
  boolean,
  number,
  select,
} from '@storybook/addon-knobs';

const story = storiesOf('Components|Button', module)
  .addDecorator(withKnobs)
  .addDecorator(withInfo)
  .addParameters({
    jest: ['index.test.tsx'],
  });

story.add(
  'Button Component',
  () => (
    <Button
      borderRadius={number('borderRadius', 0)}
      size={radios(
        'Size',
        { default: 'default', medium: 'medium', large: 'large' },
        'large'
      )}
      variant={radios(
        'Variant',
        { danger: 'danger', secondary: 'secondary', primary: 'primary' },
        'danger'
      )}
    >
      {text('Label (child)', 'Button Label', 0)}
    </Button>
  ),
  {
    info: {
      source: false,
      text: `${readme}`,
      inline: true, // Displays info inline vs click button to view
      header: false,
      propTables: [],
    },
  }
);
