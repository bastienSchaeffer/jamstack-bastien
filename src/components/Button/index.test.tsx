import React from 'react';
import { render } from '../../util/test';
import Button from './';
import serializer from 'jest-emotion';
expect.addSnapshotSerializer(serializer);

describe('button', () => {
  test('Button renders', () => {
    const { getByText } = render(<Button>Button label</Button>);

    expect(getByText(`Button label`)).toMatchSnapshot();
  });

  test('<Button> renders with props', () => {
    const { getByText } = render(
      <Button m={2} px={[1, 2, 3]} size="medium">
        Button label
      </Button>
    );

    expect(getByText(`Button label`)).toMatchSnapshot();
    expect(getByText(`Button label`)).toHaveStyle(`margin: 8px`);
  });
});
