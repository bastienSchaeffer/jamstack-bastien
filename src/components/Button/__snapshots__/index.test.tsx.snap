// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button <Button> renders with props 1`] = `
.emotion-0 {
  border: 0;
  outline: 0;
  box-shadow: 0 3px 10px rgba(0,0,0,.16);
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  border-radius: 4px;
  font-size: 16px;
  padding: 8px 16px;
  margin: 8px;
  padding-left: 4px;
  padding-right: 4px;
}

@media screen and (min-width:576px) {
  .emotion-0 {
    padding-left: 8px;
    padding-right: 8px;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

<button
  class="emotion-0"
>
  Button label
</button>
`;

exports[`button Button renders 1`] = `
.emotion-0 {
  border: 0;
  outline: 0;
  box-shadow: 0 3px 10px rgba(0,0,0,.16);
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  border-radius: 4px;
}

<button
  class="emotion-0"
>
  Button label
</button>
`;
